plugins/inputs/sqlserver/azuresqlpoolqueries.go

```diff
@@ -0,0 +1,512 @@
+package sqlserver
+
+import (
+	_ "github.com/denisenkom/go-mssqldb" // go-mssqldb initialization
+)
+
+//------------------------------------------------------------------------------------------------
+//------------------ Azure Sql Elastic Pool ------------------------------------------------------
+//------------------------------------------------------------------------------------------------
+const sqlAzurePoolResourceStats = `
+DECLARE @ErrorMessage AS nvarchar(500)
+IF SERVERPROPERTY('EngineEdition') <> 5 BEGIN /*not Azure SQL DB*/
+	SET @ErrorMessage = 'Telegraf - Connection string Server:'+ @@SERVERNAME + ',Database:' + DB_NAME() +' is not an Azure SQL DB. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+IF NOT EXISTS (SELECT 1 FROM sys.database_service_objectives WHERE database_id = DB_ID() AND elastic_pool_name IS NOT NULL) BEGIN
+	SET @ErrorMessage = 'Telegraf - Database:' + DB_NAME() +' is not in an elastic pool. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+
+SELECT TOP(1)
+   'sqlserver_pool_resource_stats' AS [measurement]
+  ,REPLACE(@@SERVERNAME,'\',':') AS [sql_instance]
+  ,(SELECT [elastic_pool_name] FROM sys.database_service_objectives WHERE database_id = DB_ID()) AS [elastic_pool_name]
+  ,[snapshot_time]
+  ,cast([cap_vcores_used_percent] as float) AS [avg_cpu_percent]
+  ,cast([avg_data_io_percent] as float) AS [avg_data_io_percent]
+  ,cast([avg_log_write_percent] as float) AS [avg_log_write_percent]
+  ,cast([avg_storage_percent] as float) AS [avg_storage_percent]
+  ,cast([max_session_percent] as float) AS [max_session_percent]
+  ,cast([max_data_space_kb]/1024. as int) AS [storage_limit_mb]
+  ,cast([avg_instance_cpu_percent] as float) AS [avg_instance_cpu_percent]
+  ,cast([avg_allocated_storage_percent] as float) AS [avg_allocated_storage_percent]
+FROM 
+  sys.dm_resource_governor_resource_pools_history_ex
+WHERE 
+  [name] = 'SloSharedPool1'
+ORDER BY
+  [snapshot_time] DESC;
+`
+
+const sqlAzurePoolResourceGovernance = `
+DECLARE @ErrorMessage AS nvarchar(500)
+IF SERVERPROPERTY('EngineEdition') <> 5 BEGIN /*not Azure SQL DB*/
+	SET @ErrorMessage = 'Telegraf - Connection string Server:'+ @@SERVERNAME + ',Database:' + DB_NAME() +' is not an Azure SQL DB. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+IF NOT EXISTS (SELECT 1 FROM sys.database_service_objectives WHERE database_id = DB_ID() AND elastic_pool_name IS NOT NULL) BEGIN
+	SET @ErrorMessage = 'Telegraf - Database:' + DB_NAME() +' is not in an elastic pool. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
```

Comment:
Can we merge the 2 checks and return 1 message instead of 2? Aka combine the checks together and say that " This is not an Azure SQL database in an elastic pool " rather than potentially raising 2 separate messages if there is a wrong connection string each time. This is true for many of the queries below

Reply:
Updated code to merge error messages.